src/components/workspace/BuildProject/BuildProject.tsx

```diff
@@ -102,23 +102,29 @@ const BuildProject: FC<Props> = ({ projectId, contract, updateContract }) => {
 
   const contractsToDeploy = () => {
     return projectFiles
-      .filter((f) => {
-        const _fileExtension = getFileExtension(f.name || '');
-        return (
-          f.path.startsWith(`${activeProject?.path}/dist`) &&
-          ['abi'].includes(_fileExtension as string)
+      .filter((file) => {
+        const fileExtension = getFileExtension(file.name || '');
+        const isAbiFile =
+          file.path.startsWith(`${activeProject?.path}/dist`) &&
+          fileExtension === 'abi';
+
+        if (activeProject?.language === 'func') {
+          return isAbiFile;
+        }
+
+        const hasTsFile = projectFiles.some(
+          (f) => f.path === file.path.replace('.abi', '.ts'),
         );
+        return isAbiFile && hasTsFile;
       })
-      .map((f) => {
-        return {
-          id: f.id,
-          name: f.name
-            .replace('.abi', '')
-            .replace('tact_', '')
-            .replace('func_', ''),
-          path: f.path,
-        };
-      });
+      .map((file) => ({
+        id: file.id,
+        name: file.name
+          .replace('.abi', '')
```

Comment:
These strings are repeated several times over the code base, and in case there'll be a change to them, there'll be no way to ensure modification is consistent.

It'd be better to assign this `replace` action a name and encapsulate all the related strings in a single file.